### PR TITLE
Update database schema and call proc insert_subsample_for_image_full_path_v2

### PR DESCRIPTION
**Pull request by KarlLevik**
* Use database schema 1.31.0
* Use stored procedure `insert_subsample_for_image_full_path_v2`
* Add optional parameter `experiment_type`
* Test use of optional parameter

### CODE REVIEW

File: requirements_orm.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-mysql-connector-python
+mysql-connector-python<=8.0.29
```
Comment (rjgildea) [inline requirements_orm.txt:1]: Is there a specific need for this restriction?

File: tests/test_xtalimaging.py [hunk 2]
```diff
@@ -16,7 +18,8 @@ def test_xtal_imaging(testdb):
     assert sid is not None
     assert sid > 0
 
-    si_full_path = "/dls/i03/data/2018/cm14451-99/something_new.jpg"
+    secs = int(datetime.datetime.now().timestamp())
+    si_full_path = "/dls/i03/data/2018/cm14451-99/something_new_%d.jpg" % secs
```
Comment (rjgildea) [inline tests/test_xtalimaging.py]: Possibly clearer/more modern Python to use f-String here?
```suggestion
    si_full_path = f"/dls/i03/data/2018/cm14451-99/something_new_{secs}.jpg"
```

File: tests/test_xtalimaging.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import datetime
```
Comment (rjgildea) [inline tests/test_xtalimaging.py]:
```suggestion
import time
```

File: tests/test_xtalimaging.py [hunk 4]
```diff
@@ -16,7 +18,8 @@ def test_xtal_imaging(testdb):
     assert sid is not None
     assert sid > 0
 
-    si_full_path = "/dls/i03/data/2018/cm14451-99/something_new.jpg"
+    secs = int(datetime.datetime.now().timestamp())
```
Comment (rjgildea) [inline tests/test_xtalimaging.py]: Simpler to use `time.time()`?
```suggestion
    secs = int(time.time())
```

File: .azure-pipelines/azure-pipelines.yml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 variables:
-  DATABASE_SCHEMA: 1.30.0
+  DATABASE_SCHEMA: 1.31.0
```
Comment (rjgildea) [inline .azure-pipelines/azure-pipelines.yml:2]: Once merged this should (I think) trigger a PR to regenerate the sqlalchemy ORM for this database version.

File: tests/test_xtalimaging.py [hunk 6]
```diff
@@ -50,5 +53,6 @@ def test_xtal_imaging(testdb):
         position1y=687,
         position2x=473,
         position2y=744,
+        experiment_type='MAD',
```
Comment (rjgildea) [inline tests/test_xtalimaging.py]: I guess you're not using pre-commits, as I would have thought `black` would have changed this to double quotes?
```suggestion
        experiment_type="MAD",
```

**Review by rjgildea — CHANGES_REQUESTED** [review]
I think this looks fine, other than a couple of minor things that could be simplified/reformatted.

File: requirements_orm.txt (hunk 1 above)
Comment (KarlLevik) [reply requirements_orm.txt:1]: Yes, unfortunately it doesn't work with 8.0.30 - note the red cross on the previous commit. There might be some incompatibility with MariaDB or its default settings. See also https://github.com/ispyb/py-ispyb/pull/206 where they ran into the same/similar problem.

File: .azure-pipelines/azure-pipelines.yml (hunk 5 above)
Comment (KarlLevik) [reply .azure-pipelines/azure-pipelines.yml:2]: That is also my impression. Fingers crossed.

File: tests/test_xtalimaging.py (hunk 2 above)
Comment (KarlLevik) [reply tests/test_xtalimaging.py]: Good point, done!

**Review by rjgildea — APPROVED** [review]

**Pull request merged**